render/render.go

```diff
@@ -478,14 +478,84 @@ func (c *Canvas) SavePNG(filename string) error {
 // Render renders a layout tree to a canvas.
 func Render(root *layout.LayoutBox, width, height int) *Canvas {
 	canvas := NewCanvas(width, height)
-	// Default white background
-	canvas.Clear(color.RGBA{255, 255, 255, 255})
+	
+	// CSS 2.1 §14.2: The background of the root element becomes the canvas background.
+	// First try to find the body's background, then fall back to root's background.
+	canvasBg := findCanvasBackground(root)
```

Comment:
Is all this background color stuff actually necessary and part of specs?  Can it be simplified and made more aligned with spec?